scripts/zones/Batallia_Downs_[S]/npcs/Thorben.lua

```diff
@@ -0,0 +1,65 @@
+-----------------------------------
+-- Area: Batallia Downs (S)
+--  NPC: Thorben
+-- Type: Quest NPC
+-- !pos  175.346,8.038,-419.244 84
+-----------------------------------
+require("scripts/globals/keyitems")
+require("scripts/globals/settings")
+require("scripts/globals/quests")
+require("scripts/globals/npc_util")
+-----------------------------------
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
+    local lostInTranslocation = player:getQuestStatus(CRYSTAL_WAR, dsp.quest.id.crystalWar.LOST_IN_TRANSLOCATION)
+    local leftMapPiece = player:hasKeyItem(dsp.ki.LEFT_MAP_PIECE)
+    local middleMapPiece = player:hasKeyItem(dsp.ki.MIDDLE_MAP_PIECE)
+    local rightMapPiece = player:hasKeyItem(dsp.ki.RIGHT_MAP_PIECE)
+
+    if lostInTranslocation == QUEST_COMPLETED then
+        player:startEvent(108)
+    elseif lostInTranslocation == QUEST_ACCEPTED and leftMapPiece and middleMapPiece and rightMapPiece then
+        player:startEvent(107)
+    elseif player:getVar("lostInTranslocationCS") == 1 and (leftMapPiece or middleMapPiece or rightMapPiece) then
+        player:startEvent(106)
+    elseif lostInTranslocation == QUEST_ACCEPTED and (leftMapPiece or middleMapPiece or rightMapPiece) then
+        player:startEvent(105)
+    elseif lostInTranslocation == QUEST_ACCEPTED and not leftMapPiece and not middleMapPiece and not rightMapPiece then
+        player:startEvent(104)
+    else
+        player:startEvent(103)
+    end
+end
+
+function onEventUpdate(player,csid,option)
+end
+
+function onEventFinish(player,csid,option)
+    if csid == 107 then
+        if player:hasKeyItem(dsp.ki.MAP_OF_GRAUBERG) then
+            npcUtil.completeQuest(player, CRYSTAL_WAR, dsp.quest.id.crystalWar.LOST_IN_TRANSLOCATION, {
+                gil = 2000,
+                xp = 2000,
+                var = "lostInTranslocationCS"
+            })
+            player:delKeyItem(dsp.ki.LEFT_MAP_PIECE)
```

Comment:
Can save three lines by breaking these deletions out of the Grauberg Map check (but still inside cs 107).

Reply:
can't believe I miss this, thank you for pointing it out  😸️